specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+  /// The default is COREWEBVIEW2_COOKIE_SAME_SITE_KIND_LAX.
+  [propget] HRESULT SameSite([out, retval] COREWEBVIEW2_COOKIE_SAME_SITE_KIND* sameSite);
+  /// Set the SameSite property.
+  [propput] HRESULT SameSite([in] COREWEBVIEW2_COOKIE_SAME_SITE_KIND sameSite);
+
+  /// The security level of this cookie. True if the client is only to return
+  /// the cookie in subsequent requests if those requests use HTTPS.
+  /// The default is false.
+  /// Note that cookie that requests COREWEBVIEW2_COOKIE_SAME_SITE_KIND_NONE but
+  /// is not marked Secure will be rejected.
+  [propget] HRESULT IsSecure([out, retval] BOOL* isSecure);
+  /// Set the IsSecure property.
+  [propput] HRESULT IsSecure([in] BOOL isSecure);
+
+  /// Whether this is a session cookie. The default is false.
+  [propget] HRESULT IsSession([out, retval] BOOL* isSession);
+}
+
+/// Creates, adds or updates, gets, or or view the cookies. The changes would
+/// apply to the context of the user profile. That is, other WebViews under the
+/// same user profile could be affected.
+[uuid(588C8A15-A28A-4FFD-926B-5E6EE7449E7C), object, pointer_default(unique)]
+interface ICoreWebView2CookieManager : IUnknown {
+  /// Create a cookie object with a specified name, value, domain, and path.
+  /// One can set other optional properties after cookie creation.
+  /// This only creates a cookie object and it is not added to the cookie
+  /// manager until you call AddOrUpdateCookie.
+  /// See ICoreWebView2Cookie for more details.
+  HRESULT CreateCookieWithDetails(
+    [in] LPCWSTR name,
+    [in] LPCWSTR value,
+    [in] LPCWSTR domain,
+    [in] LPCWSTR path,
+    [out, retval] ICoreWebView2Cookie** cookie);
+
+  /// Creates a cookie whose params matches those of the specified cookie.
+  HRESULT CreateCookie(
+    [in] ICoreWebView2Cookie* cookieParam,
+    [out, retval] ICoreWebView2Cookie** cookie);
+
+  /// Gets a list of cookies matching the specific URI.
+  /// If uri is empty string or null, all cookies under the same profile are
+  /// returned.
+  /// You can modify the cookie objects, call
+  /// ICoreWebView2CookieManager::AddOrUpdateCookie, and the changes
+  /// will be applied to the webview.
+  HRESULT GetCookies(
+    [in] LPCWSTR uri,
+    [in] ICoreWebView2GetCookiesCompletedHandler* handler);
+
+  /// Adds or updates a cookie with the given cookie data; may overwrite
+  /// cookies with matching name, domain, and path if they exist.
+  HRESULT AddOrUpdateCookie([in] ICoreWebView2Cookie* cookie);
+
+  /// Deletes a cookie whose name and domain/path pair
+  /// match those of the specified cookie.
+  HRESULT DeleteCookie([in] ICoreWebView2Cookie* cookie);
+
+  /// Deletes cookies with matching name and uri.
+  /// Cookie name is required.
+  /// If uri is specified, deletes all cookies with the given name where domain
+  /// and path match provided URI.
+  HRESULT DeleteCookies([in] LPCWSTR name, [in] LPCWSTR uri);

```

Comment:
Is there a use case for "Delete all cookies from all sites with a specific name"? The name makes sense only relative to the site. A cookie named "id" means something different on different sites. Or is this for dealing with various injected cookie scenarios, like those used for advertising or analytics? E.g. "Delete all cookies named `JSESSIONID` from all sites"?

Reply:
No. uri (or domain/path pair in `DeleteCookiesWithDomainAndPath` is required. API details updated.